File: BibliotecaUFABC/app/src/main/java/com/nintersoft/bibliotecaufabc/notification/NotificationDisplay.java
package com.nintersoft.bibliotecaufabc.notification;

import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.nintersoft.bibliotecaufabc.utilities.GlobalConstants;
import com.nintersoft.bibliotecaufabc.utilities.GlobalFunctions;

public class NotificationDisplay extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        int notificationID = intent.getIntExtra(GlobalConstants.NOTIFICATION_ID, -1);
        NotificationManager notificationManager = (NotificationManager)context.getSystemService(Context.NOTIFICATION_SERVICE);
        if (notificationManager != null)
            notificationManager.notify(notificationID,
                    GlobalFunctions.createNotification(context, notificationID, intent.getStringExtra(GlobalConstants.NOTIFICATION_MESSAGE)));
    }
}

File: BibliotecaUFABC/app/src/main/java/com/nintersoft/bibliotecaufabc/notifications/NotificationDisplay.kt
package com.nintersoft.bibliotecaufabc.notifications

import android.app.NotificationManager
import android.content.Context
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.nintersoft.bibliotecaufabc.global.Constants
import com.nintersoft.bibliotecaufabc.global.Functions

class NotificationDisplay(private val context: Context, private val workerParams: WorkerParameters)
    : Worker(context, workerParams) {

    override fun doWork(): Result {
        (context.getSystemService(Context.NOTIFICATION_SERVICE) as? NotificationManager)?.run {
            cancelAll()
            notify(workerParams.inputData.getInt(Constants.NOTIFICATION_ID, -1),
                Functions.createRenewalNotification(workerParams.inputData.
                    getString(Constants.NOTIFICATION_MESSAGE)))
        }

        return Result.success()
    }
}